pkg/secrets/passphrase/manager.go

```diff
@@ -142,6 +142,32 @@ func NewPassphaseSecretsManager(phrase string, state string) (secrets.Manager, e
 	return sm, nil
 }
 
+// Tries to find the Passphrase first using `PULUMI_CONFIG_PASSPHRASE` then
+// `PULUMI_CONFIG_PASSPHRASE_FILE` if it is not found and defaulting to an empty string
+func getConfigPassphrase() (string, error) {
+	passphrase, isOk := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE")
+	if isOk {
+		return passphrase, nil
+	} else {
+		phraseFile, isOk := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE_FILE")
+		if isOk {
+			phraseFilePath, err := filepath.Abs(phraseFile)
+			if err != nil {
+				return "", errors.Wrap(err, "unable to detect passphrase path")
+			}
+
+			phraseDetails, err := ioutil.ReadFile(phraseFilePath)
+			if err != nil {
+				return "", errors.Wrap(err, "unable to read PULUMI_CONFIG_PASSPHRASE_FILE")
+			}
+
+			return strings.TrimSpace(string(phraseDetails)), nil
+		}
+	}
```

Comment:
Total nit - but this could all be:

```go
if passphrase, isOk := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE"); isOk {
	return passphrase, nil
}
if phraseFile, isOk := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE_FILE"); isOk {
	phraseFilePath, err := filepath.Abs(phraseFile)
	if err != nil {
		return "", errors.Wrap(err, "unable to detect passphrase path")
	}
	phraseDetails, err := ioutil.ReadFile(phraseFilePath)
	if err != nil {
		return "", errors.Wrap(err, "unable to read PULUMI_CONFIG_PASSPHRASE_FILE")
	}
	return strings.TrimSpace(string(phraseDetails)), nil
}
return "", nil
```

Reply:
You are absolutely right - much cleaned - updated